### PR TITLE
BUGFIX: Handle maximum cache lifetime of 0

### PR DESCRIPTION
**Pull request by daniellienert**
If a cached prototype had a `maximumLifetime` of 0
configured, which is meant as infinite cache
lifetime, this value is propagated upwards to all
surrounding prototypes as it is the selected
as minimum value

### CODE REVIEW

File: Neos.Fusion/Classes/Core/Cache/RuntimeContentCache.php [hunk 1]
```diff
@@ -211,11 +211,16 @@ public function preEvaluate(array &$evaluateContext, $fusionObject)
 
 
             if (isset($evaluateContext['configuration']['maximumLifetime'])) {
-                $maximumLifetime = $this->runtime->evaluate($evaluateContext['fusionPath'] . '/__meta/cache/maximumLifetime', $fusionObject);
+                $maximumLifetime = (int)$this->runtime->evaluate($evaluateContext['fusionPath'] . '/__meta/cache/maximumLifetime', $fusionObject);
 
                 if ($maximumLifetime !== null && $this->cacheMetadata !== []) {
```
[the remark this answers is not in the document]
Comment (daniellienert) [reply Neos.Fusion/Classes/Core/Cache/RuntimeContentCache.php]: Absolutely! The question is, why the tests aren't failing ..

**Comment by bwaidelich** [comment]
So lifetime of `0` should behave the same as `null`?

**Comment by kdambekalns** [comment]
Not sure how it actually _is_, but I think'd expect `null` to mean "whatever is the default" and `0` to mean "unlimited".

**Comment by jonnitto** [comment]
I‘m here with @kdambekalns

**Comment by daniellienert** [comment]
Thats correct. IMHO there are three cases:
`0` --> infinite for the cache element where it is defined, should not be propagated to surrounding prototype cache segments
`null` --> undefined, should not be propagated to surrounding prototype cache segments
`int > 0` --> Should lower the cache lifetime of surrounding segments if it is lower

**Comment by daniellienert** [comment]
@kdambekalns , @bwaidelich , @jonnitto  - fixed the typecast some time ago. Would you please take another look.

**Pull request merged**